lib/module.js

```diff
@@ -86,10 +86,18 @@ module.exports = async function gtmModule (_options) {
     this.options.head = {}
   }
   this.options.head.script = this.options.head.script || []
-  this.options.head.script.push({
-    hid: options.scriptId,
-    innerHTML: script
-  })
+  if (options.nonce) {
+    this.options.head.script.push({
+      hid: options.scriptId,
+      nonce: options.nonce,
+      innerHTML: script
+    })
+  } else {
+    this.options.head.script.push({
+      hid: options.scriptId,
```

Comment:
Could we clean this up by not using if/else and instead something like `nonce: options.nonce || undefined`? 🤔

Reply:
I was thinking the same thing but unsure if having `<script nonce=undefined />` is the same as `<script />` although in both cases the value of `nonce` would be equal (undefined). I guess I'll try the more concise syntax then

Reply:
I think if the value is undefined, it won't be set at all 😋

Reply:
I think the value does get set into undefined since my test was failing but either way it shouldnt be a problem. 

Many Thanks!

Reply:
Sounds like having value set to "undefined" could cause issues?

Reply:
I'm not aware of the case where nonce value being set to undefined is causing an issue.

I don't think the CSP mechanism can tell the difference between a non existent `nonce` attribute and a `nonce` attribute with the value of undefined.

CMIIW

Reply:
I'm not 100% sure but based just on the understanding of the DOM, the value of "undefined" won't be treated differently from an actual hash value. It's not like "undefined" is treated specially and will result in the attribute being ignored.

Reply:
Hmm, I suppose the safer (and more idiomatic) way is to use the initial mechanism that I've implemented.

Reply:
The problem with your initial change is that you are repeating yourself so now if something needs to be changed in this code, it needs to be done in two places. You can instead do:

```js
    this.options.head.script.push({
      ...(options.nonce ? { nonce: options.nonce } : {}),
      hid: options.scriptId,
      innerHTML: script
    })
```

Reply:
Oh, that looks great. I didn't thought of that, Thanks!